charms/worker/k8s/src/charm.py

```diff
@@ -149,10 +151,47 @@ def __init__(self, *args):
         )
 
         self.framework.observe(self.on.update_status, self._on_update_status)
+        self.framework.observe(self.on.feature_relation_changed, self._on_feature_relation_changed)
+
```

Comment:
so, with the reconciler pattern in this charm, registering for each event isn't necessary.  You're going to see every event the charm can possible observe in the `self._reconcile` method.  

It's likely best to have this `_on_feature_relation_changed` method be named `_resolve_feature_relations` and called from `self._reconcile`

Reply:
oh, also this should only be run on the "leader-control-plane" unit.  no where else